### DISCUSSION TITLE
Fallback locale for dialects

### DISCUSSION THREAD

**Discussion by riccardolardi** [amannn/next-intl#985] [General]
Does `next-intl` support something similar to https://github.com/i18next/next-i18next?tab=readme-ov-file#fallback-locales ?

My use case is that the default locale should be `de-CH` (swiss german) and users with locale `de` should be forwarded / should fall back to `de-CH`. Same with `it-CH` (swiss italian) and `fr-CH` (swiss french) that should also accept their "native" locale

**Comment by amannn** [comment]
`next-intl` doesn't support a "fallback locale", but the most appropriate locale should always be used regardless.

I've just added some tests in https://github.com/amannn/next-intl/commit/e6231f9bfe0a5f4c5b8bbcdb28be05fef68a1836 to verify this—hope this helps!

**Comment by loking** [comment]
The lack of native dialect fallback in next-intl is a real gap — if your de-CH file is missing a key, you'd expect it to resolve from de before falling back to en, but it doesn't.

`next-intl-localechain` handles exactly this with deep key-level merging:

```javascript
import { createLocaleChain } from 'next-intl-localechain';

const chain = createLocaleChain({
  'de-CH': ['de-CH', 'de', 'en'],
  'pt-BR': ['pt-BR', 'pt', 'en'],
});
```

Ships with 75 pre-configured CLDR-based chains so you don't have to map dialect hierarchies yourself.
